src/turmoil/Turmoil.ts

```
@@ -0,0 +1,389 @@
+        // Begin with 13 neutral delegates in the reserve
+        for (let i = 0; i < 13; i++) {
+            this.delegate_reserve.push("NEUTRAL");   
+        }
+
+        // Init the global event dealer
+        this.globalEventDealer = new GlobalEventDealer();
+        this.globalEventDealer.initGlobalEvents(game);
+        this.initGlobalEvent(game);
+    }
+
+    public initGlobalEvent(game: Game) {
+        // Draw the first global event to setup the game
+        this.commingGlobalEvent = this.globalEventDealer.draw();
+        if (this.commingGlobalEvent !== undefined) {
+            this.sendDelegateToParty("NEUTRAL", this.commingGlobalEvent.revealedDelegate, game);
+        }
+        this.distantGlobalEvent = this.globalEventDealer.draw();
+        if (this.distantGlobalEvent !== undefined) {
+            this.sendDelegateToParty("NEUTRAL", this.distantGlobalEvent.revealedDelegate, game);
+        }
+    }
+
+    // Function to return a party by name
+    public getPartyByName(partyName: PartyName): IParty | undefined {
+        const party = this.parties.find((party) => party.name === partyName);
+        if (party) {
+            return party;
+        }
+        else {
+            return undefined;
+        }
+    }
+
+    // Use to send a delegate to a specific party
+    public sendDelegateToParty(player: Player | "NEUTRAL", partyName: PartyName, game: Game): void {
+        const party = this.getPartyByName(partyName);
+        if (party) {
+            if (player != "NEUTRAL" && this.lobby.has(player)) {
+                this.lobby.delete(player);
+            }
+            else {
+                const index = this.delegate_reserve.indexOf(player);
+                if (index > -1) {
+                    this.delegate_reserve.splice(index, 1);
+                }
+            }
+            party.sendDelegate(player, game);
+            this.checkDominantParty(party);
+        }
+        else {
+            console.error("Party not found");
```

In the spirit of failing fast I would throw an error in these situations.